packages/@react-types/table/src/index.d.ts

```diff
@@ -122,20 +122,20 @@ export interface SpectrumColumnProps<T> extends ColumnProps<T> {
   hideHeader?: boolean
 }
 
-export type RowElement = ReactElement<RowProps>;
+export type RowElement<T> = ReactElement<RowProps<T>>;
 export interface TableBodyProps<T> extends Omit<AsyncLoadable, 'isLoading'> {
   /** The contents of the table body. Supports static items or a function for dynamic rendering. */
-  children: RowElement | RowElement[] | ((item: T) => RowElement),
+  children: RowElement<T> | RowElement<T>[] | ((item: T) => RowElement<T>),
   /** A list of row objects in the table body used when dynamically rendering rows. */
   items?: Iterable<T>,
   /** The current loading state of the table. */
   loadingState?: LoadingState
 }
 
-export interface RowProps {
+export interface RowProps<T> {
   // treeble case? Unsupported props for now
-  // /** A list of child item objects used when dynamically rendering row children. */
-  // childItems?: Iterable<T>,
+  /** A list of child item objects used when dynamically rendering row children. */
+  childItems?: Iterable<T>,
```

Comment:
Note that this API addition will be shown when we release but it won't actually do anything without the feature flag. Perhaps I could mark it as alpha?

Reply:
i think that would be a good idea? or at least something to indicate that it won't work without the feature flag